src/extension-support/extension-manager.js

```diff
@@ -391,23 +391,22 @@ class ExtensionManager {
             }
             break;
         default:
-            if (!blockInfo.opcode) {
-                throw new Error('Missing opcode for block');
-            }
-
-            blockInfo.func = blockInfo.func ? this._sanitizeID(blockInfo.func) : blockInfo.opcode;
+            if (blockInfo.opcode) {
```

Comment:
Any reason for rearranging this conditional (from the previous structure of falsey check and early exit)?

Reply:
It's kind of silly... but this isn't legal JavaScript:
```js
default:
    if (!blockInfo.opcode) {
        throw new Error('Missing opcode for block');
    }
    const funcName = "stuff"; // it's not legal to make a new `const` here
```

You can do it if you introduce a new scope with `{}` which sort of naturally led me to the arrangement proposed here...

Reply:
Hmmmm okay. I think I'd prefer adding the new block of scope only because it's easier to read an early exit vs. keeping track of which block of code corresponds to which part of the if/else etc. Esp. since there are nested ifs involved here.